include/envoy/secret/secret.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * Secret contains certificate chain and private key.
```

Comment:
That diverges quite a lot from the API, in which Secret is opaque container for various secrets (certificates, session ticket keys, etc.).

Could we perhaps make this opaque interface that returns `Protobuf::Message&` which can then be used for config generation?

Reply:
Added a new virtual method and removed TLS certificate related methods.